pysph/examples/rb_rotation_matrices/stack_of_cylinders.py

```diff
@@ -0,0 +1,239 @@
+        xc3, yc3 = (xc2 + self.cylinder_diameter + self.cylinder_spacing, yc2)
+        xc4, yc4 = (xc3 + self.cylinder_diameter + self.cylinder_spacing, yc3)
+        xc5, yc5 = (xc4 + self.cylinder_diameter + self.cylinder_spacing, yc4)
+
+        x_five_1, y_five_1 = np.concatenate(
+            (xc1, xc2, xc3, xc4, xc5)), np.concatenate((yc1, yc2, yc3, yc4,
+                                                        yc5))
+        x_five_2, y_five_2 = x_five_1, y_five_1 + 2. * self.cylinder_diameter
+        x_five_3, y_five_3 = x_five_2, y_five_2 + 2. * self.cylinder_diameter
+        x_five, y_five = np.concatenate((x_five_1, x_five_2,
+                                         x_five_3)), np.concatenate(
+                                             (y_five_1, y_five_2, y_five_3))
+
+        x, y = np.concatenate((x_six, x_five)), np.concatenate((y_six, y_five))
+
+        # create body_id
+        no_particles_one_cylinder = len(xc1)
+        total_bodies = 3 * 5 + 3 * 6
+
+        body_id = np.array([], dtype=int)
+        for i in range(total_bodies):
+            b_id = np.ones(no_particles_one_cylinder, dtype=int) * i
+            body_id = np.concatenate((body_id, b_id))
+
+        return x, y, body_id
+
+    def geometry(self):
+        import matplotlib.pyplot as plt
+        # please run this function to know how
+        # geometry looks like
+        xc, yc, body_id = self.create_cylinders_stack()
+        xt, yt = self.create_dam()
+        xw, yw = self.create_wall()
+
+        plt.scatter(xc, yc)
+        plt.scatter(xt, yt)
+        plt.scatter(xw, yw)
+        plt.axes().set_aspect('equal', 'datalim')
+        print("done")
+        plt.show()
+
+    def post_step(self, solver):
+        t = solver.t
+        dt = solver.dt
+        T = self.wall_time
+        if (T - dt / 2) < t < (T + dt / 2):
+            for pa in self.particles:
+                if pa.name == 'wall':
+                    pa.y += 14 * 1e-2
+
+    def post_process(self):
+        if len(self.output_files) == 0:
+            return
+
+        from pysph.solver.utils import iter_output
+        files = self.output_files
+        print(len(files))
+        t = []
+        system_x = []
+        system_y = []
+        for sd, array in iter_output(files, 'cylinders'):
+            _t = sd['t']
+            if _t > self.wall_time:
+                t.append(_t)
+                # get the system center
+                cm_x = 0
+                cm_y = 0
+                for i in range(array.n_body[0]):
+                    cm_x += array.xcm[3 * i] * array.total_mass[i]
+                    cm_y += array.xcm[3 * i + 1] * array.total_mass[i]
+                cm_x = cm_x / np.sum(array.total_mass)
+                cm_y = cm_y / np.sum(array.total_mass)
+
+                system_x.append(cm_x / self.dam_length)
+                system_y.append(cm_y / self.dam_length)
+
+        import matplotlib.pyplot as plt
```

Comment:
Why is the above not done?  The `geometry` function has this but is never called.  The problem with doing the above is that it is possible that the example is run via ssh and then this will crash.  For example, ssh to a machine and try importing matplotlib's pyplot and it will crash because the backend may be set to Qt.  By doing the above, you ensure that the post processing will always work when mpl is available regardless of the system etc.